mrp_lot_number_propagation/models/mrp_production.py

```diff
@@ -0,0 +1,148 @@
+        "move_raw_ids.move_line_ids.qty_done",
+        "move_raw_ids.move_line_ids.lot_id",
+    )
+    def _compute_propagated_lot_producing(self):
+        for order in self:
+            order.propagated_lot_producing = False
+            move_with_lot = order.move_raw_ids.filtered(
+                lambda o: o.propagate_lot_number
+            )
+            line_with_sn = move_with_lot.move_line_ids.filtered(
+                lambda l: (
+                    l.lot_id
+                    and l.product_id.tracking == "serial"
+                    and tools.float_compare(
+                        l.qty_done, 1, precision_rounding=l.product_uom_id.rounding
+                    )
+                    == 0
+                )
+            )
+            if len(line_with_sn) == 1:
+                order.propagated_lot_producing = line_with_sn.lot_id.name
+
+    @api.onchange("bom_id")
+    def _onchange_bom_id_lot_number_propagation(self):
+        self.is_lot_number_propagated = self.bom_id.lot_number_propagation
+
+    def action_confirm(self):
+        res = super().action_confirm()
+        self._set_lot_number_propagation_data_from_bom()
+        return res
+
+    def _set_lot_number_propagation_data_from_bom(self):
+        """Copy information from BoM to the manufacturing order."""
+        for order in self:
+            order.is_lot_number_propagated = order.bom_id.lot_number_propagation
+            for move in order.move_raw_ids:
+                move.propagate_lot_number = move.bom_line_id.propagate_lot_number
+
+    def _post_inventory(self, cancel_backorder=False):
+        self._create_and_assign_propagated_lot_number()
+        return super()._post_inventory(cancel_backorder=cancel_backorder)
+
+    def _create_and_assign_propagated_lot_number(self):
+        for order in self:
+            if not order.is_lot_number_propagated or order.lot_producing_id:
+                continue
+            finish_moves = order.move_finished_ids.filtered(
+                lambda m: m.product_id == order.product_id
+                and m.state not in ("done", "cancel")
+            )
+            if finish_moves and not finish_moves.quantity_done:
+                lot = self.env["stock.production.lot"].create(
+                    {
+                        "product_id": order.product_id.id,
+                        "company_id": order.company_id.id,
+                        "name": order.propagated_lot_producing,
+                    }
+                )
+                order.with_context(lot_propagation=True).lot_producing_id = lot
+
+    def write(self, vals):
+        for order in self:
+            if (
+                order.is_lot_number_propagated
+                and "lot_producing_id" in vals
+                and not self.env.context.get("lot_propagation")
+            ):
+                raise UserError(
+                    _(
+                        "Lot/Serial number is propagated from a component, "
```

Comment:
```suggestion
                        "Serial number is propagated from a component, "
```

Reply:
I named it this way to stick with the std lot/sn form: 
![image](https://user-images.githubusercontent.com/5315285/196930986-fdb22c80-abc3-4540-8479-ab3ada1bc6f5.png)
And if later we add the support for production lot we won't need to change the string+translations.